### ISSUE TITLE
Track Meta tasks for GA

### ISSUE THREAD

**Issue by VijayanB** [opensearch-project/k-NN#49]
Copy task list from https://github.com/opensearch-project/opensearch-plugins/issues/41
Here is a meta issue for all plugins to track items which have to be taken care of between RC1 and GA for all the plugins.

- [x]  Plugin branching strategy 
- [x]  Backward compatibility testing https://github.com/opensearch-project/k-NN/issues/17
- [X]  ~~DCO checks From infra team(@peterzhuamazon ), DCO check will not be included for GA release~~
- [x]  Broken links check
- [x]  Standardize labels
- [x]  README.md, etc
- [x]  Release Tags
- [x]  Release notes
- [x]  Public Documentation
- [x]  Clean up branches
- [x]  Issue and PR templates
- [x]  CI workflows
- [ ]  Plugins dependencies from Maven -> only after [this](https://github.com/opensearch-project/opensearch-plugins/issues/45) ticket is closed.
- [x]  Test coverage reporting
- [X]  License header checks https://github.com/opensearch-project/k-NN/pull/41
- [x]  Plugin artifact names
- [x]  Branch protection rules https://github.com/opensearch-project/opensearch-plugins/issues/52 -> requires admin privilege

**Comment by VijayanB** [comment]
Closing this since corresponding opened items are tracked in separate ticket.
